WebApp/src/test/java/ev/projects/webapp/controllers/CaseControllerTest.java

```diff
@@ -1,93 +1,158 @@
+        ResponseEntity<Optional<Case>> response = restTemplate.exchange(url, HttpMethod.GET,
+                null, new ParameterizedTypeReference<>() {});
+        Optional<Case> responseCase = response.getBody();
+        assertEquals(response.getStatusCode(), HttpStatus.OK);
         assertNotNull(responseCase);
         assertTrue(responseCase.isPresent());
-        Case rCase = responseCase.get();
-        assertEquals(aCase, rCase);
-        assertEquals(ID, rCase.getID());
-        assertEquals(title, rCase.getTitle());
-        assertEquals(creatorUser, rCase.getCreatorUser());
+        Case aCase = responseCase.get();
+        assertEquals( "English", aCase.getTitle());
+        assertEquals( "English lessons", aCase.getDescription());
+        assertEquals( caseCountOffset + 1, aCase.getID());
+        assertNotNull(aCase.getCreatorUser());
+        assertEquals( userCountOffset + 1, aCase.getCreatorUser().getID());
+        assertEquals(username1, aCase.getCreatorUser().getName());
+    }
+
+    @Test
+    @Order(1)
+    public void createCaseTest() {
+        String title = "History";
+        String description = "Legendary stories";
+        TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username1, password1);
+        HttpEntity<Case> createRequest = new HttpEntity<>(createCase(title, description, null));
+        ResponseEntity<Case> createResponse = restTemplate.exchange(baseUrl, HttpMethod.POST,
+                createRequest, new ParameterizedTypeReference<>() {});
+        assertEquals(HttpStatus.OK, createResponse.getStatusCode());
+        assertCase(restTemplate, title, description, username1, (int) (caseCountOffset + 2));
+    }
+
+    @Test
+    @Order(2)
+    public void updateCaseTest() {
+        String url = baseUrl + "/" + (caseCountOffset + 3) + "/";
+        String title = "Future";
+        String description = "Legendary tales";
+        TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username1, password1);
+        HttpEntity<Case> updateRequest = new HttpEntity<>(createCase(title, description, null));
+        ResponseEntity<Case> createResponse = restTemplate.exchange(url, HttpMethod.PUT,
+                updateRequest, new ParameterizedTypeReference<>() {});
+        assertEquals(createResponse.getStatusCode(), HttpStatus.OK);
+        assertCase(restTemplate, title, description, username1, (int) (caseCountOffset + 2));
+    }
+
+    private void assertCase(TestRestTemplate restTemplate, String title, String description, String username, int index) {
+        ResponseEntity<List<Case>> getResponse = restTemplate.exchange(baseUrl, HttpMethod.GET, null, new ParameterizedTypeReference<>() {});
+        List<Case> cases = getResponse.getBody();
+        assertEquals(getResponse.getStatusCode(), HttpStatus.OK);
+        assertNotNull(cases);
+        assertEquals(index + 1, cases.size());
```

Comment:
Test are usually written to test functionality created by developer. No need to test database increments.

Reply:
Should be fixed with: e33fe2f1dc3881a517c7437fd07ad93f4a043dd2.